app/models/manageiq/providers/vmware/infra_manager/provision/cloning.rb

```diff
@@ -117,18 +117,13 @@ def start_clone(clone_options)
 
     [:config, :customization, :linked_clone].each { |key| vim_clone_options[key] = clone_options[key] }
 
-    [:folder, :host, :pool].each do |key|
+    %i[folder host pool snapshot].each do |key|
       ci = clone_options[key]
       next if ci.nil?
 
       vim_clone_options[key] = ci.ems_ref_obj
     end
 
-    if clone_options[:snapshot]
-      ci = clone_options[:snapshot]
-      vim_clone_options[:snapshot] = VimString.new(ci.ems_ref, ci.ems_ref_type, :ManagedObjectReference) if ci.ems_ref.present? && ci.ems_ref_type.present?
-    end
-
```

Comment:
NOTE this allows us to implement the same fix as https://github.com/ManageIQ/manageiq-providers-vmware/pull/826 without having to manually build the VimString here because we are able to include `EmsRefObjMixin` here https://github.com/ManageIQ/manageiq-providers-vmware/pull/828/files#diff-2d5e73f3d9f74251cf729c2a6b7e8afeff87a5e6ec38816ba3b6f6cac63a9690R2